velocity/src/main/java/me/leoko/advancedban/velocity/VelocityMethods.java

```diff
@@ -0,0 +1,371 @@
+import me.leoko.advancedban.velocity.command.CommandRecieverVelocity;
+import me.leoko.advancedban.velocity.event.PunishmentEvent;
+import me.leoko.advancedban.velocity.event.RevokePunishmentEvent;
+import me.leoko.advancedban.velocity.utils.LuckPermsOfflineUser;
+import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
+import ninja.leaping.configurate.ConfigurationNode;
+import ninja.leaping.configurate.objectmapping.ObjectMappingException;
+import ninja.leaping.configurate.yaml.YAMLConfigurationLoader;
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+public class VelocityMethods extends AbstractMethodInterface<ConfigurationNode> {
+
+  private final @MonotonicNonNull ProxyServer server;
+
+  private final Path dataDirectory;
+
+  private boolean luckPermsSupport;
+
+  @Inject
+  public VelocityMethods(Path dataDirectory,
+                         final @NonNull ProxyServer server) {
+    super(dataDirectory);
+    this.dataDirectory = dataDirectory;
+    this.server = server;
+
+    // LuckPerms check stuff here
+
+  }
+
+
+  @Override
+  protected ConfigurationNode loadConfiguration(Path configPath) throws IOException {
+    try (BufferedReader reader = Files.newBufferedReader(configPath, StandardCharsets.UTF_8)) {
+      YAMLConfigurationLoader loader = YAMLConfigurationLoader.builder().setPath(configPath).build();
+      return loader.load();
+    }
+  }
+
+  @Override
+  public String getFromUrlJson(String url, String key) {
+    try {
+      HttpURLConnection request = (HttpURLConnection) new URL(url).openConnection();
+      request.connect();
+
+      JsonParser jp = new JsonParser();
+      JsonObject json = (JsonObject) jp.parse(new InputStreamReader(request.getInputStream()));
+
+      String[] keys = key.split("\\|");
+      for (int i = 0; i < keys.length - 1; i++) {
+        json = json.getAsJsonObject(keys[i]);
+      }
+
+      return json.get(keys[keys.length - 1]).toString().replaceAll("\"", "");
+
+    } catch (Exception exc) {
+      return null;
+    }
+  }
+
+  @Override
+  public String getVersion() {
+    return server.getVersion().toString();
+  }
+
+  @Override
+  public String[] getKeys(Object file, String path) {
+    return new String[0];
+  }
+
+  @Override
+  public void setupMetrics() {
+    // no bstats for velocity yet
+  }
+
+  @Override
+  public boolean isBungee() {
+    return false;
+  }
+
+  @Override
+  public String clearFormatting(String text) {
+    return text.replaceAll("-\\d{1}", "");
```

Comment:
This regex pattern doesn't seem to match color codes. Are you sure this is right?